File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Method - Explicit impl 02 - Nested.cs
public class Class
{
    public interface I<T>
    {
        void M();
    }

    public class Nested1 : I<int>, I<string>
    {
        int I<int>.M() => 1;
        string I<string>.M() => "";
    }

    public class Nested2<T> : I<T>
    {
        T I<T>.M() => null;
    }
}

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Method - Explicit impl 02 - Nested.fs
module Module

let n1 = Class.Nested1()

let iInt = n1 :> Class.I<int>
iInt.M()

let iString = n1 :> Class.I<string>
iString.M()


let n2 = Class.Nested2<unit>()
let iUnit = n2 :> Class.I<unit>
iUnit.M()
